### PR TITLE
Changing selections for 2016 datasets

### PR DESCRIPTION
**Pull request by jeongsumin**
Changed selections are coming from Taiwan group selections.
Please comment If these codes have any wrong things.

### CODE REVIEW

File: NanoAODProduction/data/datasets/NanoAOD/2016/MC.RunIISummer16.rareprocess.yaml [hunk 1]
```diff
@@ -34,12 +34,6 @@ dataset:
     MC2016.TT_FCNC-aT2ZJ_zct:
         ? /TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/RunIISummer16NanoAODv5-PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/NANOAODSIM
         :   - /store/mc/RunIISummer16NanoAODv5/TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/NANOAODSIM/PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/100000
-    MC2016.TT_FCNC-aT2ZJ_zct:
-        ? /TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/RunIISummer16NanoAODv5-PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/NANOAODSIM
-        :   - /store/mc/RunIISummer16NanoAODv5/TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/NANOAODSIM/PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/100000
-    MC2016.TT_FCNC-aT2ZJ_zct:
-        ? /TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/RunIISummer16NanoAODv5-PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/NANOAODSIM
-        :   - /store/mc/RunIISummer16NanoAODv5/TT_FCNC-aT2ZJ_Tleptonic_ZToll_kappa_zct-MadGraph5-pythia8/NANOAODSIM/PUMoriond17_Nano1June2019_102X_mcRun2_asymptotic_v7-v1/100000
```
Comment (jhgoh) [inline NanoAODProduction/data/datasets/NanoAOD/2016/MC.RunIISummer16.rareprocess.yaml]: should this be dropped?

File: TopAnalysis/data/combineHLT/fcncTriLepton/2016.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+######changed by Taiwon setup(DoubleMu, DoubleEG)
```
Comment (jhgoh) [inline TopAnalysis/data/combineHLT/fcncTriLepton/2016.yaml]: Could be better to backup this 2016.yaml to different name and have new version.

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc [hunk 3]
```diff
@@ -108,6 +109,7 @@ bool FCNCTriLeptonCppWorker::isGoodElectron(const unsigned i) const {
   const double pt = in_Electrons_p4[0]->At(i);
   const double eta = in_Electrons_p4[1]->At(i);
   if ( pt < minElectronPt_ or std::abs(eta) > maxElectronEta_ ) return false;
+  if ( std::abs(eta) > 1.4442 and std::abs(eta) < 1.566 ) return false;
```
Comment (jhgoh) [inline TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: to be precise, this should be supercluster eta, not the electron eta.

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc [hunk 4]
```diff
@@ -269,40 +340,59 @@ bool FCNCTriLeptonCppWorker::analyze() {
...
-            for ( unsigned i=0; i<4; ++i ) std::swap(out_Lepton1_p4[i], out_Lepton3_p4[i]);
-          }
-          else {
-            std::swap(out_Lepton1_pdgId, out_Lepton2_pdgId);
-            for ( unsigned i=0; i<4; ++i ) std::swap(out_Lepton1_p4[i], out_Lepton2_p4[i]);
-          }
-        }
+	//Rearrange leptons for reconstruction Z boson (We want to reco. Z boson by lep2 and 3 that mass has the closet value to the Z mass)
+	if ( abs(out_Lepton1_p4[3]+out_Lepton2_p4[3]-91.2) < abs(out_Lepton1_p4[3]+out_Lepton3_p4[3]-91.2) ) {
+	    if ( abs(out_Lepton1_p4[3]+out_Lepton2_p4[3]-91.2) < abs(out_Lepton2_p4[3]+out_Lepton3_p4[3]-91.2) ) {
```
Comment (jhgoh) [inline TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: is this correct? scalar sum of two lepton masses?

File: NanoAODProduction/data/datasets/NanoAOD/2016/MC.RunIISummer16.rareprocess.yaml (hunk 1 above)
Comment (jhgoh) [reply NanoAODProduction/data/datasets/NanoAOD/2016/MC.RunIISummer16.rareprocess.yaml]: looks like this is a duplicated entry. resolved.

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc (hunk 3 above)
Comment (jhgoh) [reply TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: to be done later by @yhlee52

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc [hunk 5]
```diff
@@ -107,7 +108,9 @@ bool FCNCTriLeptonCppWorker::isVetoMuon(const unsigned i) const {
 bool FCNCTriLeptonCppWorker::isGoodElectron(const unsigned i) const {
   const double pt = in_Electrons_p4[0]->At(i);
   const double eta = in_Electrons_p4[1]->At(i);
+  const double SCeta = eta + in_Electrons_dEtaSC->At(i);
```
Comment (jhgoh) [inline TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: SCeta -> scEta

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc [hunk 6]
```diff
@@ -191,123 +197,220 @@ bool FCNCTriLeptonCppWorker::analyze() {
...
     if ( nGoodMuons     < 1 ) out_GoodLeptonCode -=  10;
     if ( nGoodElectrons < 1 ) out_GoodLeptonCode -= 100;
     for ( unsigned i=0; i<4; ++i ) {
-        if ( nGoodElectrons >= 1 ) out_Lepton1_p4[i] = in_Electrons_p4[i]->At(electronIdxs[0]);
-        if ( nGoodMuons     >= 1 ) out_Lepton2_p4[i] = in_Muons_p4[i]->At(muonIdxs[0]);
-        if ( nGoodMuons     >= 2 ) out_Lepton3_p4[i] = in_Muons_p4[i]->At(muonIdxs[1]);
+	if ( nGoodMuons     >= 1 && in_Muons_p4[0]->At(muonIdxs[0]) > LeadingLep_pt ) {
+	    out_Lepton2_p4[i] = in_Muons_p4[i]->At(muonIdxs[0]);
+	    if ( nGoodMuons     >= 2 && in_Muons_p4[0]->At(muonIdxs[1]) > RestLep_pt )
+		out_Lepton3_p4[i] = in_Muons_p4[i]->At(muonIdxs[1]);
```
Comment (jhgoh) [inline TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: wrap with {}

File: TopAnalysis/src/FCNCTriLeptonCppWorker.cc [hunk 7]
```diff
@@ -240,23 +240,17 @@ bool FCNCTriLeptonCppWorker::analyze() {
...
-	if ( nGoodElectrons >= 1 && in_Electrons_p4[0]->At(electronIdxs[0]) > RestLep_pt )
-	    out_Lepton1_pdgId = -11*in_Electrons_charge->At(electronIdxs[0]);
+	if ( nGoodMuons     >= 1 ) out_Lepton2_p4[i] = in_Muons_p4[i]->At(muonIdxs[0]);
+	if ( nGoodMuons     >= 2 ) out_Lepton3_p4[i] = in_Muons_p4[i]->At(muonIdxs[1]);
+	if ( nGoodElectrons >= 1 ) out_Lepton1_p4[i] = in_Electrons_p4[i]->At(electronIdxs[0]);
     }
+    if ( nGoodMuons     >= 1 ) out_Lepton2_pdgId = -13*in_Muons_charge->At(muonIdxs[0]);
+    if ( nGoodMuons     >= 2 ) out_Lepton3_pdgId = -13*in_Muons_charge->At(muonIdxs[1]);
+    if ( nGoodElectrons >= 1 ) out_Lepton1_pdgId = -11*in_Electrons_charge->At(electronIdxs[0]);
+
```
Comment (jeongsumin) [inline TopAnalysis/src/FCNCTriLeptonCppWorker.cc]: In different flavour channel, Pt of leading and rest leptons are checked (line 254~265) after leptons are filled in out_Lepton_p4 list.

**Pull request merged**